### ISSUE TITLE
Ability to support timeline sync

### ISSUE THREAD

**Issue by psfshr** [mastodon/mastodon#19036]
### Pitch

Add the ability for the server to support recording a “last read / viewed” position in an account’s timeline. 

### Motivation

To allow the synchronising of reading when using multiple clients, such that a user can stop reading on one client and pick up on another at the same point.

**Comment by ClearlyClaire** [comment]
The server already supports that through [timeline markers](https://docs.joinmastodon.org/methods/timelines/markers/) and the WebUI does use that API to record the position in the timeline. It doesn't currently make any use of the information for the Home timeline, though, and only uses it for the notifications timeline.

**Comment by psfshr** [comment]
Thanks. So I'd modify my suggestion to expand the WebUI use of timeline markers to include support for the Home timeline as well. I'm guessing that for apps it would need to be something implemented by the app to read/write the marker details via the API?

**Comment by ClearlyClaire** [comment]
Yeah, every client (including the WebUI) has to add support for that API.

**Comment by finestructure** [comment]
I've just come across the `markers` API and mentioned it in this thread, where I speculate this should enable cross-client timeline synchronisation: https://mastodon.social/@finestructure/109313251409555734

However, it was pointed out that client can't really adopt this, because the web client set the marker to the top continuously in its default mode (is it called fast mode?).

It seems to me that the web client should only set the top marker as a result of a user action actively scrolling/reaching the top, rather than it automatically tracking the top.

This would enable other clients to actually be using the `markers` API.

Does that make sense? Is this a good place to add this or would this better be placed in a new issue?
